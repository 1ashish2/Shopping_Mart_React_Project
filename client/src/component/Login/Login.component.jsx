import React from "react";
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";
import FormContainer from "../../constant/FormContainer.constant";
import { useDispatch } from "react-redux";
import { loginUserAction } from "../../redux/actions/authenticateActions";

const Login = () => {
  const dispatch = useDispatch();

  const submitHandler = (event) => {
    event.preventDefault();
    let email = event.target.email.value;
    let password = event.target.password.value;

    dispatch(loginUserAction({ email, password }));
  };

  return (
    <>
      <FormContainer>
        <h1>Sign In</h1>
        <Form onSubmit={submitHandler}>
          <Form.Group className="my-2">
            <Form.Label>Email Address</Form.Label>
            <Form.Control
              type="email"
              name="email"
              placeholder="Enter email"
              required
            ></Form.Control>
          </Form.Group>
          <Form.Group>
            <Form.Label>Password</Form.Label>
            <Form.Control
              type="password"
              name="password"
              placeholder="Enter password"
              required
            ></Form.Control>
          </Form.Group>
          <Button
            data-testid="button"
            type="submit"
            className="my-2"
            variant="primary"
          >
            Sign In
          </Button>
        </Form>
      </FormContainer>
    </>
  );
};

export default Login;
