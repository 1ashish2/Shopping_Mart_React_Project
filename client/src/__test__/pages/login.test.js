

import { render, cleanup } from "@testing-library/react";
import { MemoryRouter } from 'react-router-dom'
import Login from "../../pages/login.page";

jest.mock("react-redux", () => ({
    useSelector: jest.fn(),
    useDispatch: jest.fn(),
}));

afterEach(() => {
    cleanup();
})

describe("Message container Component", () => {
   
    render(<Login  />, { wrapper: MemoryRouter });
    it("matches snapshot", () => {
        const { asFragment } =  render(<Login  />, { wrapper: MemoryRouter });

        expect(asFragment()).toMatchSnapshot();
    })
    
    
    
   
})