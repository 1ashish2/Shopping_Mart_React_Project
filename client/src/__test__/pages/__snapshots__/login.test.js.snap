// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message container Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="form-container container"
  >
    <div
      class="justify-content-md-center  row"
    >
      <div
        class="col-md-6 col-12"
      >
        <h1>
          Sign In
        </h1>
        <form
          class=""
        >
          <div
            class="my-2"
          >
            <label
              class="form-label"
            >
              Email Address
            </label>
            <input
              aria-required="true"
              class="form-control"
              name="email"
              placeholder="Enter email"
              required=""
              type="email"
            />
          </div>
          <div>
            <label
              class="form-label"
            >
              Password
            </label>
            <input
              aria-required="true"
              class="form-control"
              name="password"
              placeholder="Enter password"
              required=""
              type="password"
            />
          </div>
          <button
            class="my-2 btn btn-primary"
            data-testid="button"
            type="submit"
          >
            Sign In
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
